File: quick-sort/QS.swift
func quickSort<T:Comparable>(_ a:[T]) -> [T] {
    guard a.count > 1 else {return a}

    let key = a.first!
    let headL = quickSort(a[1..<a.endIndex].filter{$0 < key})
    let tailL = quickSort(a[1..<a.endIndex].filter{$0 >= key})
    
    let result = headL + [key] + tailL
    
    return result
}

var a = [10,2,3,4,5]
print(quickSort(a))

File: quick-sort/QS.hs
--quickSort :: (Ord a) => [a] -> [a]
quickSort [] =[]
quickSort (x:y) = (quickSort $ filter (< x) y) ++ x:(quickSort $ filter (>= x) y)

main = do
  print $ quickSort [2,3,2,1,4,5,6,7,8,5,4,2,1,3,5,6,3,2,1,3,5,3]
